packages/roosterjs-editor-dom/lib/clipboard/extractClipboardEvent.ts

```diff
@@ -24,16 +26,28 @@ interface WindowForIE extends Window {
 export default function extractClipboardEvent(
     event: ClipboardEvent,
     callback: (clipboardData: ClipboardData) => void,
-    options?: ExtractClipboardEventOption
+    options?: ExtractClipboardEventOption,
+    rangeBeforePaste?: Range
 ) {
     const dataTransfer =
         event.clipboardData ||
         (<WindowForIE>(<unknown>(<Node>event.target).ownerDocument?.defaultView)).clipboardData;
 
     if (dataTransfer.items) {
         event.preventDefault();
-        extractClipboardItems(toArray(dataTransfer.items), options).then(callback);
+        extractClipboardItems(toArray(dataTransfer.items), options)
+            .then(callback)
+            .then(() => removeContents(rangeBeforePaste));
     } else {
         extractClipboardItemsForIE(dataTransfer, callback, options);
     }
 }
+
+function removeContents(range?: Range) {
+    return new Promise<void>(resolve => {
```

Comment:
why need a promise here?

Reply:
Tested again and it is not needed. and we can actually it in the same promise as the one we call the callback.